src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -79,10 +79,18 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
         }
         codeWriter.emit(" {\n")
       } else if (anonymousTypeArguments != null) {
-        val supertype = if (superinterfaces.isNotEmpty()) superinterfaces[0] else superclass
-        codeWriter.emitCode("object : %T(", supertype)
-        codeWriter.emitCode(anonymousTypeArguments)
-        codeWriter.emit(") {\n")
+
+        codeWriter.emitCode("object :")
+        val supertype = if (superclass != ANY) {
```

Comment:
won't this break if i want
```kotlin
val FOO: Any = object : Any() {
}
```
for some reason?

how about `if (superclass != ANY || superinterfaces.isEmpty())`?

Reply:
That's equivalent to
```kotlin
val FOO: Any = object {
}
```
which is what will be emitted

Reply:
oh, cool, I didn't know that was valid Kotlin. Thanks!

Reply:
actually, won't this emit `...object : {...` (extra colon), or am i missing something again?

Reply:
Oh, right, `:` would be redundant